py/server/tests/test_pyfunc_return_java_values.py

```diff
@@ -75,6 +77,32 @@ def fn(self, col) -> {np_dtype}:
                 t = empty_table(10).update("X = i").update(f"Y= foo.fn(X + 1)")
                 self.assertNotEqual(t.columns[1].data_type, dh_dtype)
 
+    def test_datetime_scalar_return(self):
+        dt_dtypes = [
+            "np.dtype('datetime64[ns]')",
+            "np.dtype('datetime64[ms]')",
+            "datetime.datetime",
+            "pd.Timestamp"
+        ]
+
+        for np_dtype in dt_dtypes:
+            with self.subTest(np_dtype=np_dtype):
+                func_decl_str = f"""def fn(col) -> {np_dtype}:"""
+                if np_dtype == "np.dtype('datetime64[ns]')":
+                    func_body_str = f"""    return pd.Timestamp(col).to_numpy()"""
+                elif np_dtype == "datetime.datetime":
+                    func_body_str = f"""    return pd.Timestamp(col).to_pydatetime()"""
+                elif np_dtype == "pd.Timestamp":
+                    func_body_str = f"""    return pd.Timestamp(col)"""
+                func_body_str = f"""    return pd.Timestamp(col).to_numpy()"""
```

Comment:
test looks broken.  This looks like it overrides other cases..

Reply:
Ah, I wasn't too careful with co-pilot.